script/release

```diff
@@ -1,43 +1,105 @@
 #!/usr/bin/env bash
 
-read -p "MAJOR: " major
-read -p "MINOR: " minor
-read -p "PATCH: " patch
-
-# creating branch
-git checkout -b release-v$major.$minor.$patch
-
-# updating version file
-sed -E -i.bak \
-    -e "s/MAJOR = [0-9]+/MAJOR = $major/g" \
-    -e "s/MINOR = [0-9]+/MINOR = $minor/g" \
-    -e "s/PATCH = [0-9]+/PATCH = $patch/g" \
-    lib/primer/view_components/version.rb
-
-# updating changelog
-sed -i.bak "/## main/ {a\
-\\
-\\
-## $major.$minor.$patch
-}" CHANGELOG.md
-
-# removing bak files
-rm CHANGELOG.md.bak
-rm lib/primer/view_components/version.rb.bak
-
-# updating Gemfile.lock
-bundle
-pushd demo
-bundle
-popd
-
-# commiting and pushing
-git add CHANGELOG.md Gemfile.lock demo/Gemfile.lock lib/primer/view_components/version.rb
-git commit -m "release v$major.$minor.$patch"
-git push origin release-v$major.$minor.$patch
-
-echo "####################################################"
-echo "Now, open a PR with this branch and merge it to main"
-echo "Then, run bundle exec rake release on main to release the gem to RubyGems"
-echo "Finally, create a GitHub release https://github.com/primer/view_components/releases/new with the changes from CHANGELOG"
-echo "####################################################"
+create_release_branch() {
+  git checkout -B release
+}
+
+update_readme() {
+  sed -i "/## main/ {a\
+  \\
+  \\
+  ## $1.$2.$3
+  }" CHANGELOG.md
+}
+
+update_ruby_version() {
+  # Update version file
+  sed -E -i.bak \
+      -e "s/MAJOR = [0-9]+/MAJOR = $1/g" \
+      -e "s/MINOR = [0-9]+/MINOR = $2/g" \
+      -e "s/PATCH = [0-9]+/PATCH = $3/g" \
+      lib/primer/view_components/version.rb
+}
+
+update_gemfiles() {
+  # Update Gemfile.lock
+  bundle
+  pushd demo
+  bundle
+  popd
+}
+
+
+update_npm() {
+  npm version --no-git-tag-version "$1.$2.$3"
+  yarn
+}
+
+add_changed_files() {
+  git add \
+    CHANGELOG.md \
+    Gemfile.lock \
+    demo/Gemfile.lock \
+    lib/primer/view_components/version.rb \
+    app/assets/javascripts \
+    package.json \
+    yarn.lock
+}
+
+commit() {
+  git commit -m "release $1.$2.$3"
+  git tag "v$1.$2.$3"
+}
+
+push() {
+  echo "####################################################"
+  echo "Now, open a PR with this branch and merge it to main"
+  echo "Then, run script/publish on main to release the gem and the NPM package"
+  echo "Finally, create a GitHub release https://github.com/primer/view_components/releases/new with the changes from CHANGELOG"
```

Comment:
I wonder if we could automate even more of this if we were okay relying the `gh` CLI?

Reply:
Yes! That's a next step in my opinion
we could use gh API to create the release and everything